components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsAdapter.kt

```diff
@@ -16,45 +16,40 @@ import mozilla.components.support.images.loader.ImageLoader
 /**
  * Function responsible for creating a `TabViewHolder` in the `TabsAdapter`.
  */
-typealias ViewHolderProvider = (ViewGroup, BrowserTabsTray) -> TabViewHolder
+typealias ViewHolderProvider = (ViewGroup) -> TabViewHolder
 
 /**
  * RecyclerView adapter implementation to display a list/grid of tabs.
- * @param delegate TabsTray.Observer registry to allow `TabsAdapter` to conform to `Observable<TabsTray.Observer>`.
+ *
+ * @param thumbnailLoader an implementation of an [ImageLoader] for loading thumbnail images in the tabs tray.
  * @param viewHolderProvider a function that creates a `TabViewHolder`.
+ * @param delegate TabsTray.Observer registry to allow `TabsAdapter` to conform to `Observable<TabsTray.Observer>`.
  */
 @Suppress("TooManyFunctions")
 open class TabsAdapter(
     thumbnailLoader: ImageLoader? = null,
-    private val viewHolderProvider: ViewHolderProvider = { parent, tabsTray ->
+    private val viewHolderProvider: ViewHolderProvider = { parent ->
         DefaultTabViewHolder(
-                LayoutInflater.from(parent.context).inflate(
-                        R.layout.mozac_browser_tabstray_item,
-                        parent,
-                        false),
-                tabsTray,
-                thumbnailLoader
+            LayoutInflater.from(parent.context).inflate(R.layout.mozac_browser_tabstray_item, parent, false),
+            thumbnailLoader
         )
     },
     delegate: Observable<TabsTray.Observer> = ObserverRegistry()
-) : RecyclerView.Adapter<TabViewHolder>(),
-    TabsTray,
-    Observable<TabsTray.Observer> by delegate {
-
-    internal lateinit var tabsTray: BrowserTabsTray
-
+) : RecyclerView.Adapter<TabViewHolder>(), TabsTray, Observable<TabsTray.Observer> by delegate {
     private var tabs: Tabs? = null
 
+    var styling: TabsTrayStyling = TabsTrayStyling()
```

Comment:
Should we make it possible to specify in the constructor?

Reply:
That was the original intent and something I'd like to do when when we get rid of `BrowserTabsTray` entirely which limits us to getting the styling attributes from the RecyclerView after we construct the `TabsAdapter`, so we have this awkward dance. I'll file a follow-up to move this to the constructor as part of the deprecation.

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/7822.

Reply:
👍